### PR TITLE
Fix keyboard nav reactivity broken after delete items in admin settings

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
We've fixed a bug where keyboard navigation stopped working after deleting a resource in the admin settings app.
We also fixed a bug where initial keyboard navigation didn't work, when no resource was selected.

This has been archived to add respective stores of the available  user, space, group resources. Due to this change decluttering was also possible:
- move the edit/save methods to the responsible components instead bubbling up to the view component
- remove excessive code paths to handle events 

## Related Issue

- Fixes https://github.com/owncloud/web/issues/10186

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-app-admin-settings/src/views/Groups.vue [hunk 1]
```diff
@@ -202,7 +202,7 @@ export default defineComponent({
       await loadResourcesTask.perform()
       loadResourcesEventToken = eventBus.subscribe('app.admin-settings.list.load', async () => {
         await loadResourcesTask.perform()
-        selectedGroups.value = []
```
Comment (JanAckermann) [inline packages/web-app-admin-settings/src/views/Groups.vue:205]: setting to new arr broke reactivity in keyboard actions composables

Comment (kulmann) [reply packages/web-app-admin-settings/src/views/Groups.vue:205]: sounds fishy 🤔

Comment (JanAckermann) [reply packages/web-app-admin-settings/src/views/Groups.vue:205]: Indeed and I don't get it. As it is a ref, should work out of the box

File: packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue [hunk 2]
```diff
@@ -142,28 +143,18 @@ import { ContextMenuQuickAction } from '@ownclouders/web-pkg'
...
-      required: true
-    },
-    selectedSpaces: {
-      type: Array as PropType<SpaceResource[]>,
-      required: true
-    }
-  },
-  emits: ['toggleSelectSpace', 'selectSpaces', 'unSelectAllSpaces'],
-  setup: function (props, { emit }) {
+  setup: function (props) {
```
Comment (dschmidt) [inline packages/web-app-admin-settings/src/components/Spaces/SpacesList.vue:157]:
```suggestion
  setup(props) {
```

File: packages/web-app-admin-settings/tests/unit/views/__snapshots__/Users.spec.ts.snap [hunk 3]
```diff
@@ -87,91 +87,7 @@ exports[`Users view > list view > renders list initially 1`] = `
...
-                  <!-- @slot Footer of the table -->
-                  <!--v-if-->
-                  <div class="oc-text-nowrap oc-text-center oc-width-1-1 oc-my-s">
-                    <p class="oc-text-muted">1 users in total</p>
-                  </div>
-                </td>
-              </tr>
-            </tfoot>
-          </table>
+          <no-content-message-stub data-v-32efd9c6="" icon="user" iconfilltype="fill"></no-content-message-stub>
```
Comment (dschmidt) [inline packages/web-app-admin-settings/tests/unit/views/__snapshots__/Users.spec.ts.snap:90]: Is this right?

**Review by dschmidt — APPROVED** [review]

File: packages/web-app-admin-settings/tests/unit/views/__snapshots__/Users.spec.ts.snap (hunk 3 above)
Comment (JanAckermann) [reply packages/web-app-admin-settings/tests/unit/views/__snapshots__/Users.spec.ts.snap:90]: noooooooo

File: packages/web-app-admin-settings/src/composables/stores/groupSettings.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+export const useGroupSettingsStore = defineStore('groupSettings', () => {
+  const groups = ref<Group[]>([])
+  const selectedGroups = ref<Group[]>([])
+
+  const setGroups = (data: Group[]) => {
+    groups.value = data
+  }
+
+  const upsertGroup = (group: Group) => {
+    const existing = unref(groups).find(({ id }) => id === group.id)
```
Comment (kulmann) [inline packages/web-app-admin-settings/src/composables/stores/groupSettings.ts:14]:
```suggestion
    const existing = unref(groups).some(({ id }) => id === group.id)
```

File: packages/web-app-admin-settings/src/composables/stores/spaceSettings.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+export const useSpaceSettingsStore = defineStore('spaceSettings', () => {
+  const spaces = ref<SpaceResource[]>([])
+  const selectedSpaces = ref<SpaceResource[]>([])
+
+  const setSpaces = (data: SpaceResource[]) => {
+    spaces.value = data
+  }
+
+  const upsertSpace = (space: SpaceResource) => {
+    const existing = unref(spaces).find(({ id }) => id === space.id)
```
Comment (kulmann) [inline packages/web-app-admin-settings/src/composables/stores/spaceSettings.ts:14]:
```suggestion
    const existing = unref(spaces).some(({ id }) => id === space.id)
```

File: packages/web-app-admin-settings/src/composables/stores/userSettings.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+export const useUserSettingsStore = defineStore('userSettings', () => {
+  const users = ref<User[]>([])
+  const selectedUsers = ref<User[]>([])
+
+  const setUsers = (data: User[]) => {
+    users.value = data
+  }
+
+  const upsertUser = (user: User) => {
+    const existing = unref(users).find(({ id }) => id === user.id)
```
Comment (kulmann) [inline packages/web-app-admin-settings/src/composables/stores/userSettings.ts:14]:
```suggestion
    const existing = unref(users).some(({ id }) => id === user.id)
```

**Review by kulmann — APPROVED** [review]

**Pull request merged**